File: src/main/Python/task_2/vehicle.py
from abc import ABC, abstractmethod


class Vehicle(ABC):
    def __init__(self, max_speed, type):
        self.max_speed = max_speed
        self.type = type

    @abstractmethod
    def get_max_speed(self):
        pass

    @abstractmethod
    def get_type(self):
        pass

File: src/main/java/task_2/Vehicle.java
package task_2;

public abstract class Vehicle {
    int maxSpeed;
    String type;

    public Vehicle(int maxSpeed, String type) {
        this.maxSpeed = maxSpeed;
        this.type = type;
    }

    public double calculateAllowedSpeed() {
        return this.maxSpeed;
    }

    public int getMaxSpeed() {
        return this.maxSpeed;
    }

    public String getType() {
        return this.type;
    }
}
